### PR TITLE
Add new `Lint/AmbiguousAssignment` cop

### LINKED ISSUE #9223: New lint cop to check mistypes in `+=` or `-=`
Inspired by https://rules.sonarsource.com/ruby/type/Bug/RSPEC-2757 and https://www.viva64.com/en/w/v588/

I think many people did mistakes like the following
```ruby
x =+ 2
```

### PR DESCRIPTION
**Pull request by fatkodima**
Closes #9223

### CODE REVIEW

File: lib/rubocop/cop/lint/suspicious_assignment.rb [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+      #
+      class SuspiciousAssignment < Base
+        include RangeHelp
+        extend AutoCorrector
+
+        MSG = 'Suspicious assignment detected. Did you mean `%<op>s`?'
+
+        SIMPLE_ASSIGNMENT_TYPES = %i[lvasgn ivasgn cvasgn gvasgn casgn].freeze
+
+        MISTYPES = {
```
Comment (bbatsov) [inline lib/rubocop/cop/lint/suspicious_assignment.rb:27]: I think something like `MISTYPED_ASSIGNMENTS" or "MISTAKES" would be a better name here.

File: lib/rubocop/cop/lint/suspicious_assignment.rb [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+module RuboCop
+  module Cop
+    module Lint
+      # This cop checks for mistyped shorthand assignments.
+      #
+      # @example
+      #   # bad
+      #   x =- y
+      #   x =+ y
+      #   x =! y
```
Comment (bbatsov) [inline lib/rubocop/cop/lint/suspicious_assignment.rb:12]: Btw, perhaps `x =* y` should be handled as well. It does work. :-)

**Comment by dvandersluis** [comment]
Maybe `AmbiguousAssignment` would be a better name?

**Comment by dvandersluis** [comment]
Also, I think we need to handle ensure that `x = -y` is allowed, since it'll parse to the same AST as `x =- y`, and is valid. The offense message should probably suggest that `x =- y` could be either `x -= y` or `x = -y` since both are valid and we can't know what they were going for.

**Comment by bbatsov** [comment]
I think that's why there's no auto-correct for the cop - it's hard to tell if that's a bug or poorly formatted code. That being said it seems to me the code simply checks in the source if the operator is right after the `=`,  so it wouldn't result in a false positive for `x = -y`. That being said, some tests to cover this and `+/*y` would be useful.

**Comment by fatkodima** [comment]
For me both names (current and `AmbiguousAssignment`) are good enough, not sure which is better.
Expanded tests.

**Comment by dvandersluis** [comment]
`AmbiguousAssignment` would be parallel with the other Lint cops (`AmbiguousOperator`, `AmbiguousBlockAssociation`, `AmbiguousRegexpLiteral`), but I'd leave it up to @bbatsov

**Comment by bbatsov** [comment]
Let's go with Ambiguous. Consistent naming is a good thing.

**Comment by fatkodima** [comment]
Changed the name.

**Comment by bbatsov** [comment]
Thanks!

**Pull request merged**